commands/curation/curationaudit.go

```diff
@@ -49,17 +50,50 @@ const (
 		"This package manager however isn't supported by this command."
 
 	TotalConcurrentRequests = 10
+
+	MinArtiMavenSupport = "7.82.0"
+	MinArtiXraySupport  = "3.92.0"
 )
 
 var CurationOutputFormats = []string{string(outFormat.Table), string(outFormat.Json)}
 
-var supportedTech = map[coreutils.Technology]func() (bool, error){
-	coreutils.Npm: func() (bool, error) { return true, nil },
-	coreutils.Maven: func() (bool, error) {
-		return clientutils.GetBoolEnvValue(utils.CurationMavenSupport, false)
+var supportedTech = map[coreutils.Technology]func(ca *CurationAuditCommand) (bool, error){
+	coreutils.Npm: func(ca *CurationAuditCommand) (bool, error) { return true, nil },
+	coreutils.Maven: func(ca *CurationAuditCommand) (bool, error) {
+		return ca.checkSupportByVersionOrEnv(coreutils.Maven, MinArtiMavenSupport, MinArtiXraySupport, utils.CurationMavenSupport)
 	},
 }
 
+func (ca *CurationAuditCommand) checkSupportByVersionOrEnv(tech coreutils.Technology, minRtVersion, minXrayVersion, envName string) (bool, error) {
+	flag, errResolveFlag := clientutils.GetBoolEnvValue(envName, false)
+	if errResolveFlag != nil {
+		log.Error(errResolveFlag)
+	}
+	if flag {
+		return true, nil
+	}
+	rtManager, serverDetails, err := ca.getRtManagerAndAuth(tech)
+	if err != nil {
+		return false, err
+	}
+	rtVersion, err := rtManager.GetVersion()
+	if err != nil {
+		return false, err
+	}
+	_, xrayVersion, err := utils.CreateXrayServiceManagerAndGetVersion(serverDetails)
+	if err != nil {
+		return false, err
+	}
+
+	xrayVersionErr := clientutils.ValidateMinimumVersion(clientutils.Xray, xrayVersion, minXrayVersion)
+	rtVersionErr := clientutils.ValidateMinimumVersion(clientutils.Artifactory, rtVersion, minRtVersion)
```

Comment:
after getting each version, we can validate the version right away, in case we can retrurn right away, no need to check and make more API requests because we got error.

Reply:
this error is not server error but a "known" error which is "version not supported" I check both arti and xray in order to present the user both products version so he can act accordingly